worker/upload.go

```diff
@@ -739,9 +729,12 @@ func (u *uploader) execute(req *sectorUploadReq, rev types.FileContractRevision)
 	span := trace.SpanFromContext(req.ctx)
 	span.AddEvent("execute")
 
+	// create a host
+	h := req.upload.mgr.hp.newHostV3(u.fcid, u.hk, u.siamuxAddr)
```

Comment:
why recreate it on every upload? is it because the metrics recorder would otherwise hang on to too much memory or something?